app/ota_client.py

```diff
@@ -113,36 +114,50 @@ def __call__(
         url = self._path_to_url(url_base, path)
 
         error_count = 0
-        try:
-            response = self._session.get(url, stream=True, timeout=10, cookies=cookies)
-            response.raise_for_status()
-
-            raw_r = response.raw
-            if raw_r.retries:
-                error_count = len(raw_r.retries.history)
-        except requests.exceptions.HTTPError:
-            msg = f"requests error: status_code={response.status_code},{url=} ({error_count})"
-            raise OtaErrorRecoverable(msg)
-        except requests.exceptions.Timeout as e:
-            msg = f"requests timeout: {e},{url=} ({error_count})"
-            raise OtaErrorRecoverable(msg)
-        except requests.exceptions.ConnectionError as e:
-            msg = f"requests connection error: {e},{url=} ({error_count})"
-            raise OtaErrorRecoverable(msg)
-        except requests.exceptions.ChunkedEncodingError:
-            msg = f"requests ChunkedEncodingError: {url=} ({error_count})"
-            raise OtaErrorRecoverable(msg)
-        except requests.exceptions.RequestException as e:
-            # all unspecific errors go here
-            msg = f"requests failed: {e!r}"
-            raise OtaErrorRecoverable(msg)
-
-        # prepare hash
-        hash_f = sha256()
-        with open(dst, "wb") as f:
-            for data in response.iter_content(chunk_size=self.CHUNK_SIZE):
-                hash_f.update(data)
-                f.write(data)
+
+        while True:
+            try:
+                response = self._session.get(
+                    url, stream=True, timeout=10, cookies=cookies
+                )
+                response.raise_for_status()
+
+                raw_r = response.raw
+                if raw_r.retries:
+                    error_count += len(raw_r.retries.history)
+
+                # prepare hash
+                hash_f = sha256()
+                with open(dst, "wb") as f:
+                    try:
+                        for data in response.iter_content(chunk_size=self.CHUNK_SIZE):
+                            hash_f.update(data)
+                            f.write(data)
+                        break
+                    except Exception:  # catch all exceptions
+                        error_count += 1
+                        if error_count > self.RETRY_COUNT:
+                            logger.exception("error: iter_content")
+                            raise
+                        time.sleep(self.BACKOFF_MAX)
```

Comment:
If I understand correctly, here we sleep `self.BACKOFF_MAX` for every retry, is that right?

Reply:
Yes, when iter_content raises, sleep self.BACKOFF_MAX(=10s) every time.

Reply:
Is it too long for the first retry on `iter_content`? Or should we use a backoff function to generate the for every retry delay here?

Reply:
get_backoff_time has been implemented.